app/repos/people.js

```diff
@@ -35,24 +36,49 @@ const { personDto } = require('../dto/person')
 
 /**
  * @param owners
- * @param transaction
+ * @param [transaction]
+ * @param  [transaction2]
  * @returns {Promise<CreatedPersonDao[]>}
+ *
  */
 const createPeople = async (owners, transaction) => {
   if (!transaction) {
     return sequelize.transaction(async (t) => createPeople(owners, t))
   }
+  const unmanagedTransaction = await sequelize.transaction()
 
   const createdPeople = []
 
   try {
     for (const owner of owners) {
-      const person = await sequelize.models.person.create({
+      const createProperties = {
         first_name: owner.firstName,
         last_name: owner.lastName,
         birth_date: owner.dateOfBirth ?? owner.birthDate,
         person_reference: createRegistrationNumber()
-      }, { transaction })
+      }
+
+      let person
+
+      try {
+        person = await sequelize.models.person.create(createProperties, { transaction: unmanagedTransaction })
+        await unmanagedTransaction.commit()
+      } catch (e) {
+        if (e instanceof UniqueConstraintError) {
+          await unmanagedTransaction.rollback()
+          const duplicatePersonReference = e.fields.person_reference
+          let personReference = duplicatePersonReference
+
+          while (personReference === duplicatePersonReference) {
+            personReference = createRegistrationNumber()
+          }
+
+          person = await sequelize.models.person.create({ ...createProperties, person_reference: personReference }, { transaction })
+        } else {
+          transaction.rollback()
```

Comment:
There shouldn't be any need to rollback the main transaction here since that will automatically happen when the 'throw' takes it out of scope. However, you may need to rollback the unmanagedTransaction - not 100% sure without testing it.

Reply:
Just trying, you're right, I put this in as on a previous iteration was using a second nested query.